### ISSUE TITLE
Flutter will die soon

### ISSUE THREAD

**Issue by revenantcoder** [flutter/flutter#84070]
After working on flutter for about a month, I am a little disappointed. The way they handle widgets is irregular without any particular pattern. Sometimes they use static property of a class, sometimes enums, sometimes named-constructor and sometimes inherited class as a parameter. WHY ?? Lets take a simple example of basic widget. If I want to change the color of a textbutton then this is the code

TextButton(
                    style: TextButton.styleFrom(
                    primary: Colors.purple,
                  ),
and why not just 
Textbutton(
primary: Colors.purple
)
and why I need 6 named constructor for EdgeInsets -> (all, fromLTRB, symmetic, only, etc ) when one constructor was sufficient EdgeInsets(L,R,T,B). These are the example of basic widgets, wondering what flutter team did with more advanced widgets.

**Comment by stuartmorgan** [comment]
Closing as this does not describe a feature request or a bug. For general discussion, try one of the resources at flutter.dev/community

**Comment by revenantcoder** [comment]
Why the hell you closed this issue.. if you dont know the answer to my query, then get the hell out of here @stuartmorgan

**Comment by revenantcoder** [comment]
![Screenshot_2021-06-06-16-23-23-10](https://user-images.githubusercontent.com/65244841/120921868-94210580-c6e3-11eb-8406-94a44648a864.jpg)

**Comment by abhikommen** [comment]
@revenantcoder  Little do you know how flutter process and render your widget.. little do you know how flutter hot reloads your widget soo efficiently... first understand how things works then reach to any conclusion.. Happy Coding..

**Comment by revenantcoder** [comment]
I have a simple ques
TextButton(
style: TextButton.styleFrom(
primary: Colors.purple,
),
and why not just
Textbutton(
primary: Colors.purple
)

**Comment by abhikommen** [comment]
watch this.. https://www.youtube.com/watch?v=996ZgFRENMs&ab_channel=Flutter and it its not enough then I am always here for you.. 
btw one question ? this is your first time with mobile development ??

**Comment by revenantcoder** [comment]
@abhikommen 
yes first time with mobile but why should i consider flutter for just mobiles.. when they offers multiple platforms (android, iOS, mac, windows, linux(desktop), and web)
Ok i'll reply you back after watching the video link u shared with me.

**Comment by abhikommen** [comment]
@revenantcoder  And those 6 named constructors of EdgeInsets are there for your convenience.. 
Okay you tell me, `EdgeInsets.fromLTRB(8, 8, 8, 8)` , `EdgeInsets.all(8)`  which one makes more sense to you ???

**Comment by revenantcoder** [comment]
I think EdgeInsets(L,R,T,B) is more convenient than useless 6 named constuctors. Which increases the flutter sdk size and then app size later..

**Comment by abhikommen** [comment]
@revenantcoder Yup... flutter is multiplatform.. I asked for mobile development specially bcz if you were from "native" ios/android background then I would have explained you with a different approach... nvm..

**Comment by stuartmorgan** [comment]
I closed this issue because, as I said, the issue tracker is not a Q&A forum, but for specific purposes. I pointed you to other resources where it would be appropriate to ask questions about the design rationale.

Please see https://github.com/flutter/flutter/blob/master/CODE_OF_CONDUCT.md before engaging here again.

**Comment by abhikommen** [comment]
@revenantcoder choices are subjective... Btw, it definitely make more sense to use `EdgeInsets.all(8)` over EdgeInsets.fromLTRB(8, 8, 8, 8) .. Writing easy to read code is every developer's responsibility.  Anyways, you will learn it soon..
And.. Umm, I hate to break it for you bro but sdk never ever ever ever.... ever shifts along with your app code... Its just the sdk classes/functions that you call/imports into your code ships when you build your app.. that too into compiled form.. 


```
 const EdgeInsets.all(double value)
    : left = value,
      top = value,
      right = value,
      bottom = value;
```


That's all it takes to add `.all` name constructor to EdgeInsets class or in SDK.... yk all it adds around 131 bytes to the whole sdk.. just for your knowledge my this comment is around 1441 bytes in size.. LMAO

In fact if you use code obfuscation techniques like progaurd then you can even reduce your app size by shipping only specific method from a specific class to your project..

TL;DR, you don't have enough knowledge about the framework ( neither I do ) yet you had reached to the conclusion... As a developer its not what you should you.. You should thoroughly examine the framework first, not just flutter but any other as well then make such judgements.. Happy Coding Brother...

**Comment by Hixie** [comment]
(This issue was locked due to the way some of the conversation here violated our [code of conduct](https://github.com/flutter/flutter/blob/master/CODE_OF_CONDUCT.md), notably the part that says "Be kind. Be courteous".)
